scripts/make_gitrev.sh

```diff
@@ -35,10 +35,11 @@ EOF
 }
 trap cleanup EXIT
 
-# Check if git is installed and we are in a git repo.
-git rev-parse --git-dir > /dev/null 2>&1
+# Compute git dir path
+ZFS_GIT_DIR=$(realpath -e "$(dirname "$0")/../.git")
```

Comment:
It wouldn't hurt for the comment to explain why this is needed.  The  case of nested git repo's was new to me, and it's kind of surprising that it defaults to the outermost repo.

Reply:
> It wouldn't hurt for the comment to explain why this is needed.

I've expanded the comment, it should be clearer now.

> The case of nested git repo's was new to me, and it's kind of surprising that it defaults to the outermost repo.

And I understand why the comment is needed now :)
You were initially right, it default to the innermost repo: `git` traverse up from PWD, until it find a working git dir. But when using a git archive, this git dir won't exists, and if the archive is build inside another repo, `git describe` will use this outer repo's git dir. By forcing to use the expected git dir, it will fail to open it when in an archive thus generated an "unknown" version rather than a wrong one.